File: find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  let index = -1;
  index = haystack.indexOf(needle);
  return index;
  for (let i = 0; i < haystack.length; i++) {
    let j = i;
    let found = true;
    for (let char of needle) {
      if (j === haystack.length) {
        found = false;
        break;
      }
      if (haystack[j] !== char) {
        found = false;
        break;
      }
      j++;
    }
    if (found) {
      index = i;
      break;
    }
  }
  return index;
};

const testCases = [
  {
    haystack: "sadbutsad",
    needle: "sad",
  },
  {
    haystack: "leetcode",
    needle: "leeto",
  },
];

for (const testCase of testCases) {
  console.log(strStr(testCase.haystack, testCase.needle));
}

File: find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index = -1
        index = haystack.find(needle)
        return index
        for i in range(len(haystack)):
            j = i
            found = True
            for c in needle:
                if j == len(haystack):
                    found = False
                    break
                if c != haystack[j]:
                    found = False
                    break
                j += 1
            if found:
                index = i
                break

        return index


solution = Solution()
test_cases = [
    {"haystack": "sadbutsad", "needle": "sad"},
    {"haystack": "leetcode", "needle": "leeto"},
]

for test_case in test_cases:
    index = solution.strStr(test_case.get("haystack"), test_case.get("needle"))
    print(index)
